### PR TITLE
Issue hbz lobid 162 oai pmh

### PR DESCRIPTION
**Pull request by dr0i**
See #162.

### CODE REVIEW

File: app/models/Search.java [hunk 1]
```diff
@@ -263,14 +263,18 @@ public boolean doingScrollScanNow() {
...
+		boolean RDF_XML = serialization.equals(Serialization.RDF_XML);
 		try {
 			long lastTime = Calendar.getInstance().getTimeInMillis();
-			SearchResponse searchResponse = startInitialResponse();
+			SearchResponse searchResponse = startInitialResponse(scroll);
 			final SearchHits hits = getTotalHits(searchResponse);
 			if (JSON_LD)
 				getMessageOut().write("[");
+			if (RDF_XML)
+				getMessageOut().write("<root>");
```
Comment (fsteeg) [inline app/models/Search.java]: Does this yield valid RDF-XML? The root element looks very different from our usual RDF-XML requests (which contain namespace declarations in a root element called `rdf:RDF`), see:

`curl -L --header "Accept: application/rdf+xml" "http://lobid.org/resource?q=ehrenfeld"`

Also, it seems even with `scroll`, in our current deployment on production, we get a root element, see:

`curl -L --header "Accept: application/rdf+xml" "http://lobid.org/resource?q=ehrenfeld&scroll=true"`

You committed this to your branch a few days before I fixed https://github.com/lobid/lodmill/issues/463 so maybe this is no longer needed?

File: app/models/Search.java [hunk 2]
```diff
@@ -323,8 +329,13 @@ private SearchHits getTotalHits(final SearchResponse response) {
...
 
-	private SearchResponse startInitialResponse() {
-		final QueryBuilder queryBuilder = createQuery();
+	private SearchResponse startInitialResponse(final String scroll) {
+		QueryBuilder queryBuilder = createQuery();
+		if (scroll.matches("\\d{8}")) {
+			final QueryBuilder changedSinceQuery =
+					new LobidResources.ChangedSinceQuery().build(scroll);
+			queryBuilder = boolQuery().must(queryBuilder).must(changedSinceQuery);
+		}
```
Comment (fsteeg) [inline app/models/Search.java]: I think this should go into the `createQuery` method, where the other special parameters are handled and added to the resulting full query.

File: app/models/Search.java (hunk 1 above)
Comment (dr0i) [inline app/models/Search.java]: You should validate the results with e.g. 'xmllint' for xml and 'rapper' for rdf.
The validation shows that it's ok to wrap the '<rdf:RDF>' entries in whatever element while it's not ok to lack a root element (as it is in our current production - I don't know where you see a root element there?)

File: app/models/Search.java (hunk 2 above)
Comment (dr0i) [inline app/models/Search.java]: Done. Though I don' like, in general, the usage of global variables. I like to have the used varibles in the method definition, as parameters. Makes them methods easier to understand at a glance, IMO. But may be a matter of taste.

File: app/models/Search.java (hunk 1 above)
Comment (fsteeg) [inline app/models/Search.java]: You're right, with `scroll` the response in production is not well-formed, and the wrapped `<root>` works fine with rapper.

**Pull request merged**